### ISSUE TITLE
[Key Vault] Ensure port numbers are preserved in vault URLs parsed from IDs

### ISSUE THREAD

**Issue by mccoyp** [Azure/azure-sdk-for-java#28971]
See https://github.com/Azure/azure-sdk-for-python/issues/24446 for issue context, and https://github.com/Azure/azure-sdk-for-python/pull/24466 for Python's fix.

**Comment by nagyesta** [comment]
Hi @mccoyp and @vcolin7 ,
thank you both for looking into this! I might be wrong, but I think this should be a related issue: https://github.com/Azure/azure-sdk-for-java/issues/24508
Since that was merged, I didn't see issues with port numbers being removed with the Java client.

**Comment by mccoyp** [comment]
Wow @nagyesta, it looks like you're right! Thank you for having kept us honest on more than one occasion 🙂

I opened issues for each of our primary languages to make sure the bug wasn't more widespread, even though I did see you mention in https://github.com/Azure/azure-sdk-for-python/issues/24446 that Java and .NET worked as expected. #24508 is a duplicate that led to a fix and tests though, so we can close this issue. Thank you again for pointing out the original issue, and for following up on this one!
